src/membership/members.rs

```diff
@@ -70,6 +98,8 @@ pub struct Profile<T: Trait> {
     pub entry: EntryMethod<T>,
     pub suspended: bool,
     pub subscription: Option<T::SubscriptionId>,
+    pub controller_account: T::AccountId,
```

Comment:
Can you add more comments here about what different fields are. This is particularly important in order to distinguish role of different accounts. Also the `roles` field deserves to be explained, given its complex structure.